app/bundles/CoreBundle/Security/Permissions/CorePermissions.php

```diff
@@ -360,6 +360,11 @@ public function hasEntityAccess($ownPermission, $otherPermission, $ownerId = 0)
             return false;
         }
 
+        // assume logged user is owner for new entity
+        if (null === $ownerId) {
+            $ownerId = $user;
+        }
```

Comment:
I'm worried about this assumption. It's not secure to assume it. Will this PR work without this?

Reply:
It will not work because third parameter of method set user from getCreatedBy().  But this method return null for new entity. 

 https://github.com/mautic/mautic/pull/11024/files#diff-6128249c053d16b80e1641a97fb28c2ebe64864136fab01b959aa127f37139c8R225

We can add it directly  something like 

` $options['data']->getCreatedBy() ?? $this->userHelper->getUser()`

But it fix it just in email., not in stages...

Reply:
I see. It sucks that in a method `hasEntityAccess` the entity is missing from the params. We cannot ask `$entity->isNew()` which would make it safer. And this method is used on many places so it would require quite a refactoring effort to create new method with the entity param and deprecate this one. Although if you are in the refactoring mood today, go for it ;)

If not, I'd suggest to set the owner before this method is called in the specific controller. I really feel uncomfortable about what pandora box this change would open.

Reply:
I've made method hasAccessByEntity 
https://github.com/mautic/mautic/pull/11024/files#diff-0e305e9a7ea014329a43647e8c8a9ac2efb20ea783bd57aacd8d7b5d13806530R353